src/main/java/seedu/address/model/transaction/Transaction.java

```diff
@@ -0,0 +1,92 @@
+package seedu.address.model.transaction;
+
+import seedu.address.model.person.Person;
+import seedu.address.model.person.UniquePersonList;
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+
+import java.util.Objects;
+import java.util.Set;
+
+public class Transaction {
+
+    private final Person payer;
+    private final Double amount;
+    private final String description;
+    private final UniquePersonList payees;
+
+    public Transaction(Person payer, Double amount, String description, UniquePersonList payees) {
+        this.payer = payer;
+        this.amount = amount;
+        this.description = description;
+        this.payees = payees;
+    }
+
+    public Transaction(Person payer, Double amount, String description, Set<Person> payeesToAdd) {
+        UniquePersonList payees = new UniquePersonList();
+        for(Person p: payeesToAdd) {
+            try {
+                payees.add(p);
+            } catch (DuplicatePersonException e) {
+                System.out.println("Duplicate person" + p.getName() + " not added to list of payees");
+            }
+        }
+
+        this.payer = payer;
+        this.amount = amount;
+        this.description = description;
+        this.payees = payees;
+    }
+
+    public Person getPayer() {
+        return payer;
+    }
+
+    public Double getAmount() {
+        return amount;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public UniquePersonList getPayees() {
+        return payees;
+    }
+
+    @Override
+    public boolean equals(Object other) {
```

Comment:
We should had an unique **Id** for each transaction. Because it could be possible that the same transaction (with same persons, same amout) occurs a second time, so this function "equals" would work. It would treat the two transactions as same.